Cql/Cql.CqlToElm/InvocationBuilder.cs

```diff
@@ -27,270 +22,265 @@ internal record ResolveResult<T>(T Result, int Cost, string? Error)
-                return new(SystemLibrary.LongToDecimal.Build(locatorContext, argument), 1, null);
-            if (argumentType == SystemTypes.IntegerType && to == SystemTypes.QuantityType)
-                return new(SystemLibrary.IntegerToQuantity.Build(locatorContext, argument), 2, null);
-            if (argumentType == SystemTypes.LongType && to == SystemTypes.QuantityType)
-                return new(SystemLibrary.LongToQuantity.Build(locatorContext, argument), 2, null);
-            if (argumentType == SystemTypes.DecimalType && to == SystemTypes.QuantityType)
-                return new(SystemLibrary.DecimalToQuantity.Build(locatorContext, argument), 2, null);
-            if (argumentType == SystemTypes.DateType && to == SystemTypes.DateTimeType)
-                return new(SystemLibrary.DateToDateTime.Build(locatorContext, argument), 2, null);
-            if (argumentType == SystemTypes.CodeType && to == SystemTypes.ConceptType)
-                return new(SystemLibrary.CodeToConcept.Build(locatorContext, argument), 1, null);
-
-            // TODO: There is still ValueSet to List, but it's not clear which function to call,
-            // ToList<T>(T) would probably create a list with a single element, the valueset, but
-            // that is not the intent.
-
-            // TODO: interval promotion https://cql.hl7.org/03-developersguide.html#promotion-and-demotion
-
-            // List demotion https://cql.hl7.org/03-developersguide.html#promotion-and-demotion
-            if (argumentType is ListTypeSpecifier && to is not ListTypeSpecifier)
+        internal static TypeSpecifier ReplaceGenericTypes(TypeSpecifier type, IDictionary<string, TypeSpecifier> replacements)
+        {
+            if (type is ParameterTypeSpecifier generic)
             {
-                var singleton = SystemLibrary.SingletonFrom.Call(Provider, locatorContext, argument);
-                var intermediate = BuildImplicitCast(singleton, to, out newAssignments);
-                return intermediate with { Cost = intermediate.Cost + 5 };
+                if (replacements.TryGetValue(generic.parameterName, out var resolvedType))
+                    return resolvedType;
+                else throw new ArgumentException($"Generic type {generic.parameterName} does not have a replacement defined.", nameof(replacements));
             }
+            else if (type is ListTypeSpecifier listType)
+                return ReplaceGenericTypes(listType.elementType, replacements).ToListType();
+            else if (type is IntervalTypeSpecifier intervalType)
+                return ReplaceGenericTypes(intervalType.pointType, replacements).ToIntervalType();
+            else return type;
+        }
 
-            // TODO: interval demotion https://cql.hl7.org/03-developersguide.html#promotion-and-demotion
+        internal ConversionResult<Expression> TryGenericReplacement(TypeSpecifier operandType, Expression argument, IDictionary<string, TypeSpecifier> replacements)
```

Comment:
Renamed to `ReplaceGeneric*` instead of `Try*`.  The rationale was that the `ConversionResult` type has a `Success` property on it (along with other things), but I agree with you on the expected usage associated with the word `Try`.